storage/id_alloc_test.go

```diff
@@ -215,27 +215,9 @@ func TestAllocateWithStopper(t *testing.T) {
 		log.Fatal(err)
 	}
 
-	var wg sync.WaitGroup
-	wg.Add(10)
-	ch := make(chan struct{})
+	stopper.Stop()
 
-	stopper.RunWorker(func() {
-		<-ch // wait for signal to start.
-		for i := 0; i < 10; i++ {
-			go func() {
-				_, err := idAlloc.Allocate()
-				// We expect all allocations to fail.
-				if err != nil {
-					wg.Done()
-				} else {
-					t.Fatal("unexpected success")
-				}
-			}()
-		}
-	})
-
-	// Stop the stopper pre-emptively, then signal the waiting worker to try allocations.
-	go stopper.Stop()
-	close(ch)
-	wg.Wait()
+	if _, err := idAlloc.Allocate(); err == nil || !strings.Contains(err.Error(), "system is draining") {
+		t.Fatal("unexpected success")
```

Comment:
This will be a confusing error message if the strings.Contains check ever starts failing. Make it two separate failure cases.